lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -416,6 +416,46 @@ getLocationInfoAsString(const SmallPtrSet<Operation *, 8> &ops) {
   return sstr.str();
 }
 
+/// Given string \p name, generate and return a new name if it conflicts with
+/// any keyword or any other name in the set \p recordNames. Use the int \p
+/// nextGeneratedNameID as a counter for suffix.
+std::string resolveKeywordConflict(const StringRef &origName,
```

Comment:
You can change this to return a StringRef to avoid a copy (the stringref would point into the string set, by having this function do the insertion).  You can change "const StringRef&" to just `StringRef origName`.

StringRef resolveKeywordConflict(StringRef origName, ...

Reply:
Made the changes. Now the function updates the StringSet and returns the iterator corresponding to the new entry.